coalib/results/Diff.py

```diff
@@ -351,14 +351,17 @@ def __bool__(self):
     def delete_line(self, line_nr):
         """
         Mark the given line nr as deleted. The first line is line number 1.
+        Raises an error if line nr doesn't exist in the diff.
         """
+        line_nr = 0 if (len(self._file) < line_nr) else line_nr
```

Comment:
Any reason for setting it to 0 :sweat_smile:  IMO you should raise an error here only.